Sequence_analyzer_tools/additional_modules/filter_fastq.py

```diff
@@ -0,0 +1,44 @@
+from typing import Tuple
+
+
+def gc_content_calculator(sequence: str) -> float:
+    length_sequence = len(sequence)
+    if length_sequence == 0:
+        return 0.0
+    gc_count = (
+        sequence.count("G")
+        + sequence.count("C")
+        + sequence.count("g")
+        + sequence.count("c")
+    )
+    if sequence:
+        return gc_count / length_sequence * 100
+    else:
+        return 0
```

Comment:
Функция хорошо работает, корректно обрабатывает пустые строки и учитывает разные регистры оснований. Подсчет gc_count включает вызов count для каждого символа, что требует нескольких проходов по строке. Можно попробовать оптимизировать, например, с помощью генератора. Также часть с ```if sequence``` мне кажется немного избыточной
```suggestion
def gc_content_calculator(sequence: str) -> float:
    length_sequence = len(sequence)
    if length_sequence == 0:
        return 0.0
    gc_count = sum(1 for base in sequence if base.upper() in {"G", "C"})
    return gc_count / length_sequence * 100
```